### PR TITLE
add CORS

### LINKED ISSUE #158: apidef: enable CORS configuration on an API Definition object
See: https://github.com/TykTechnologies/tyk-operator/blob/master/docs/api_definitions.md

Tyk Operator currently does not support CORS.

- [ ] Update apidefinition_types.go
- [ ] Update [feature table](https://github.com/TykTechnologies/tyk-operator/blob/master/docs/api_definitions.md)
- [ ] Provide examples 
- [ ] Ensure it is working for both Tyk CE and Tyk Pro deployments

### PR DESCRIPTION
**Pull request by gernest**
- [x] Update apidefinition_types.go
 - [x] Update feature table
 - [x] Provide examples
- [x] Tested
  - [x] tyk ce
  - [x] tyk pro

Closes #158

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -128,3 +128,7 @@ bundle: manifests
 .PHONY: bundle-build
 bundle-build:
 	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
+
+cross-build-image:
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -mod=vendor -a -o manager.exe main.go
+	docker build -f cross.Dockerfile . -t ${IMG}
```
Comment (asoorm) [inline Makefile:134]: nit: https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

File: cross.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM gcr.io/distroless/static:nonroot
+WORKDIR /
+COPY manager.exe manager
+USER nonroot:nonroot
+
+ENTRYPOINT ["/manager"]
```
Comment (asoorm) [inline cross.Dockerfile:6]: nit: newline

File: api/v1alpha1/apidefinition_types.go [hunk 3]
```diff
@@ -529,16 +528,17 @@ type ProxyTransport struct {
...
-	Enable             bool         `json:"enable"`
-	AllowedOrigins     []string     `json:"allowed_origins"`
-	AllowedMethods     []HttpMethod `json:"allowed_methods"`
-	AllowedHeaders     []string     `json:"allowed_headers"`
-	ExposedHeaders     []string     `json:"exposed_headers"`
-	AllowCredentials   bool         `json:"allow_credentials"`
-	MaxAge             int          `json:"max_age"`
-	OptionsPassthrough bool         `json:"options_passthrough"`
-	Debug              bool         `json:"debug"`
+	Enable             bool         `json:"enable,omitempty"`
```
Comment (asoorm) [inline api/v1alpha1/apidefinition_types.go:533]: comments above each field will be propagated to oas3 schema.
This is useful for IDEs - as it forms documentation.

example: https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/#define-the-api

Screenshot 2020-11-12 at 21 25 55

File: Makefile (hunk 1 above)
Comment (gernest) [reply Makefile]: done f159b5e

File: cross.Dockerfile (hunk 2 above)
Comment (gernest) [reply cross.Dockerfile]: done f159b5e

File: api/v1alpha1/apidefinition_types.go (hunk 3 above)
Comment (gernest) [reply api/v1alpha1/apidefinition_types.go]: done dc16e57

**Comment by gernest** [comment]
@asoorm PTAL. I addressed all the comments and tested on both `ce` and `pro` edition.

File: Makefile [hunk 4]
```diff
@@ -128,3 +128,7 @@ bundle: manifests
 .PHONY: bundle-build
 bundle-build:
 	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
+
+cross-build-image:
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -mod=vendor -a -o manager.exe main.go
```
Comment (asoorm) [inline Makefile:133]: Can we rename this to `manager` or `manager.linux` and add to `.gitignore`?

Comment (gernest) [reply Makefile]: done 9e0b89e

**Pull request merged**